### ISSUE TITLE
Google Pay is not shown again on dismiss of the error dialog

### ISSUE THREAD

**Issue by jaejoong-yoo-gtg** [Adyen/adyen-android#1800]
## Description
In a situation where `DropInServiceResult.Error` with an error dialog is passed to SDK after submitting Google Pay payments, Google Pay is not shown again on dismiss of the error dialog. And, it just stays in a blank screen. It navigates back to the payment methods list if I click on the back button.
If it's hard to launch Google Pay again, it should go to the payment methods list at least, instead of staying in the middle of nowhere.

## Steps to Reproduce
1. I am able to consistently reproduce this issue: Yes
2. Steps to reproduce the issue:
    1. Select Google Pay from payment methods list in DropIn
    2. Pay with a card that could fall into a case where DropInService would send an error result

## Code Snippets
This is a test code that will reproduce the issue all the time.
```
class PaymentsCheckoutService : DropInService() {
    override fun onSubmit(state: PaymentComponentState<*>) {
        launch {
            delay(500)
            sendResult(DropInServiceResult.Error(errorDialog = ErrorDialog(message = "error")))
        }
    }
}
```

## Integration Information
1. Server-side integration: Advanced flow
2. Client-side integration: Drop-in
3. SDK version: com.adyen.checkout:drop-in:5.6.0
4. Android version(s) where issue occurs: API 34
5. Device model(s) where issue occurs: Pixel 7a

## Additional Context
N/A

**Comment by OscarSpruit** [comment]
Hi @jaejoong-yoo-gtg, thank you for reporting this bug! We were able to reproduce it and will look into a fix. We'll keep you updated on the progress.

**Comment by OscarSpruit** [comment]
Hi @jaejoong-yoo-gtg, we have released [5.9.0](https://github.com/Adyen/adyen-android/releases/tag/5.9.0) which addresses the issue. Feel free to reopen the issue if anything.
